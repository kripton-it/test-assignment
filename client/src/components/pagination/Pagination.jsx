import React from 'react';

import * as Styles from './styles';

const Pagination = ({ currentPage, pageNumber, onNextClick, onPrevClick }) => {
  const prevButton = (
    <Styles.PrevButton onClick={onPrevClick}>
      {'<'}
    </Styles.PrevButton>
  );
  const nextButton = (
    <Styles.NextButton onClick={onNextClick}>
      {'>'}
    </Styles.NextButton>
  );
  return (
    <Styles.PaginationContainer>
      {currentPage > 1 ? prevButton : null}
      {currentPage < pageNumber ? nextButton : null}
    </Styles.PaginationContainer>
  );
};

export default Pagination;