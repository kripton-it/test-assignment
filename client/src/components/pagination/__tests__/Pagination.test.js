import React from "react";
import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';

import Pagination from "../Pagination";

const props1 = {
  currentPage: 1,
  pageNumber: 5,
  onNextClick: () => { },
  onPrevClick: () => { }
};

const props2 = {
  currentPage: 5,
  pageNumber: 5,
  onNextClick: () => { },
  onPrevClick: () => { }
};

const props3 = {
  currentPage: 3,
  pageNumber: 5,
  onNextClick: () => { },
  onPrevClick: () => { }
};

describe("Pagination component:", () => {
  it("expect to render for currentPage === 1", () => {
    const wrapper = shallow(<Pagination {...props1} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("expect to render for currentPage === pageNumber", () => {
    const wrapper = shallow(<Pagination {...props2} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("expect to render for currentPage !== 1 and !== pageNumber", () => {
    const wrapper = shallow(<Pagination {...props3} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
