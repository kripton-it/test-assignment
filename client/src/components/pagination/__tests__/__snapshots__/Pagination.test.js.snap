// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component: expect to render for currentPage !== 1 and !== pageNumber 1`] = `
<PaginationContainer>
  <PrevButton
    onClick={[Function]}
  >
    &lt;
  </PrevButton>
  <NextButton
    onClick={[Function]}
  >
    &gt;
  </NextButton>
</PaginationContainer>
`;

exports[`Pagination component: expect to render for currentPage === 1 1`] = `
<PaginationContainer>
  <NextButton
    onClick={[Function]}
  >
    &gt;
  </NextButton>
</PaginationContainer>
`;

exports[`Pagination component: expect to render for currentPage === pageNumber 1`] = `
<PaginationContainer>
  <PrevButton
    onClick={[Function]}
  >
    &lt;
  </PrevButton>
</PaginationContainer>
`;
